src/util/util_errors.h

```diff
@@ -164,6 +164,8 @@ enum sssd_errors {
     ERR_INI_PARSE_FAILED,
     ERR_INI_ADD_SNIPPETS_FAILED,
 
+    ERR_TLS_FAILURE,
```

Comment:
I think the name is way too generic for our use case.
I would propose something like ERR_TLS_HANDSHAKE_INTERRUPTED.